Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm

```diff
@@ -384,16 +411,30 @@ + (void)deleteLocalAuthenticatorCredentialWithID:(NSData *)credentialID
 + (void)deleteLocalAuthenticatorCredentialWithGroupAndID:(NSString *)group credential:(NSData *)credentialID
 {
 #if ENABLE(WEB_AUTHN)
+    BOOL useAlternateKeychainAttribute = NO;
+#if USE(APPLE_INTERNAL_SDK)
+    useAlternateKeychainAttribute = shouldUseAlternateKeychainAttribute();
+#endif
     auto deleteQuery = adoptNS([[NSMutableDictionary alloc] init]);
     [deleteQuery setDictionary:@{
         (__bridge id)kSecClass: bridge_id_cast(kSecClassKey),
-        (__bridge id)kSecAttrApplicationLabel: credentialID,
         (__bridge id)kSecUseDataProtectionKeychain: @YES,
         (__bridge id)kSecAttrSynchronizable: (id)kSecAttrSynchronizableAny
     }];
     updateQueryForGroupIfNecessary(deleteQuery.get(), group);
 
+    if (useAlternateKeychainAttribute)
+        [deleteQuery setObject:credentialID forKey:(__bridge id)kSecAttrAlias];
+    else
+        [deleteQuery setObject:credentialID forKey:(__bridge id)kSecAttrApplicationLabel];
+
     SecItemDelete((__bridge CFDictionaryRef)deleteQuery.get());
+
+    if (useAlternateKeychainAttribute) {
+        [deleteQuery removeObjectForKey:bridge_cast(kSecAttrAlias)];
+        [deleteQuery setObject:credentialID forKey:bridge_cast(kSecAttrApplicationLabel)];
+        SecItemDelete((__bridge CFDictionaryRef)deleteQuery.get());
```

Comment:
Should you log the status if/when this fails?

Reply:
We haven't logged delete failures so far so I didn't want to deviate from that.